Sources/LeafKit/Data/TemplateData.swift

```diff
@@ -110,153 +115,112 @@ public struct LeafData: Equatable, LeafDataRepresentable, ExpressibleByDictionar
+            case .data(let d):        return String(data: d, encoding: .utf8)
+            case .lazy(let lazy):     return lazy().string
+            default:                  return nil
         }
     }
 
     /// Attempts to convert to `Int` or returns `nil`.
     public var int: Int? {
         switch storage {
-        case .int(let i):
-            return i
-        case .string(let s):
-            return Int(s)
-        case .lazy(let lazy):
-            return lazy().int
-        default:
-            return nil
+            case .int(let i):     return i
+            case .string(let s):  return Int(s)
+            case .lazy(let lazy): return lazy().int
+            default:              return nil
         }
     }
 
     /// Attempts to convert to `Double` or returns `nil`.
     public var double: Double? {
         switch storage {
-        case .int(let i):
-            return Double(i)
-        case .double(let d):
-            return d
-        case .string(let s):
-            return Double(s)
-        case .lazy(let lazy):
-            return lazy().double
-        default:
-            return nil
+            case .int(let i):     return Double(i)
+            case .double(let d):  return d
+            case .string(let s):  return Double(s)
+            case .lazy(let lazy): return lazy().double
+            default:              return nil
         }
     }
 
     /// Attempts to convert to `Data` or returns `nil`.
     public var data: Data? {
         switch storage {
-        case .data(let d):
-            return d
-        case .string(let s):
-            return s.data(using: .utf8)
-        case .lazy(let lazy):
-            return lazy().data
-        case .int(let i):
-            return i.description.data(using: .utf8)
-        case .double(let d):
-            return d.description.data(using: .utf8)
-        case .array(let arr):
-            var data = Data()
-
-            for i in arr {
-                switch i {
-                case .null: break
-                default:
-                    guard let u = i.data else {
-                        return nil
-                    }
-
-                    data += u
+            case .data(let d):            return d
+            case .string(let s):          return s.data(using: .utf8)
+            case .lazy(let lazy):         return lazy().data
+            case .int(let i):             return i.description.data(using: .utf8)
+            case .double(let d):          return d.description.data(using: .utf8)
+            case .dictionary:             return nil
+            case .bool:                   return nil
+            case .null:                   return nil
+            case .array(let arr):
+                var data = Data()
+                for i in arr {
+                    if i.isNull { break }
+                    let u = i.data
+                    guard u != nil else { return nil }
+                    data += u!
```

Comment:
Minor change to switch case here for readability - possibly slightly more efficient not to have a nested switch? Unclear